### PR TITLE
Turns Insights Today's Stats into multiple rows on iPhone

### LINKED ISSUE #290: Make Insights Today stats multiple rows on iPhone
Copy the row styling from the rows under the graph for the today stats on Insights when on iPhone. Like:

When you're on an iPad still continue to show the 1x4 grid.

cc: @isaackeyet

### LINKED ISSUE #303: Selecting today's Insights does not switch segmented control to Days on iPad
1. Launch StatsDemo on iPad.
2. Click any one of the cells in the Today's Stats section in Insights.

Expected: Segmented control switches to "Days" and highlights the proper selection under the graph.
Actual: Segmented control incorrectly stays on Insights yet the view switches to Days and the proper selection under the graph is made.

### PR DESCRIPTION
**Pull request by astralbodies**
Closes #290
Fixes #303, #292

Copies over some of logic and the cell used in the stats graph section for use in Insights Today's Stats. On the iPhone the layout is modified to use multiple rows rather than the 2x2 like previous or the 1x4 grid like on iPad.

iPad (original layout) - 
ios simulator screen shot sep 1 2015 1 21 48 pm

iPhone (new layout) -
ios simulator screen shot sep 1 2015 1 24 39 pm

Testing (using StatsDemo) - 
* View today's stats on iPhone, iPad and verify they are the same as the web
* Try rotation, refreshing, etc.

Needs Review: @jleandroperez, @jancavan
cc: @daniloercoli

### CODE REVIEW

**Comment by jancavan** [comment]
@astralbodies downloaded the latest version of StatsDemo but I don't see the new layout for Today's Stats on my iPhone? I still see the 2x2 one.

**Comment by daniloercoli** [comment]
Seems good with me.
The new layout (list layout) is always available on the iPhone. Even when I rotate the iphone to landscape. Is that on purpose?

File: WordPressCom-Stats-iOS/InsightsTableViewController.m [hunk 1]
```diff
@@ -322,10 +321,28 @@ - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath
                 }
             }
         }
-    } else if ([identifier isEqualToString:InsightsTableTodaysStatsDetailsCellIdentifier] || [identifier isEqualToString:InsightsTableTodaysStatsDetailsiPadCellIdentifier]) {
+    } else if ([identifier isEqualToString:InsightsTableTodaysStatsDetailsiPadCellIdentifier]) {
         if ([self.statsTypeSelectionDelegate conformsToProtocol:@protocol(WPStatsSummaryTypeSelectionDelegate)]) {
             [self.statsTypeSelectionDelegate viewController:self changeStatsSummaryTypeSelection:StatsSummaryTypeViews];
         }
+    } else if (statsSection == StatsSectionInsightsTodaysStats && IS_IPAD == NO && [self.statsTypeSelectionDelegate conformsToProtocol:@protocol(WPStatsSummaryTypeSelectionDelegate)]) {
```
Comment (jleandroperez) [inline WordPressCom-Stats-iOS/InsightsTableViewController.m]: We're SO gonna need to review all of the `IS_IPAD`, app-wide!

File: WordPressCom-Stats-iOS/InsightsTableViewController.m [hunk 2]
```diff
@@ -322,10 +321,28 @@ - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath
...
+            case 1:
+                [self.statsTypeSelectionDelegate viewController:self changeStatsSummaryTypeSelection:StatsSummaryTypeViews];
+                break;
+            case 2:
+                [self.statsTypeSelectionDelegate viewController:self changeStatsSummaryTypeSelection:StatsSummaryTypeVisitors];
+                break;
+            case 3:
+                [self.statsTypeSelectionDelegate viewController:self changeStatsSummaryTypeSelection:StatsSummaryTypeLikes];
+                break;
+            case 4:
```
Comment (jleandroperez) [inline WordPressCom-Stats-iOS/InsightsTableViewController.m]: This is me, not saying anything about enums! :stuck_out_tongue_winking_eye:

File: WordPressCom-Stats-iOS/InsightsTableViewController.m [hunk 3]
```diff
@@ -679,6 +703,51 @@ - (void)configureTodaysStatsCell:(InsightsTodaysStatsTableViewCell *)cell
...
+    cell.selectedIsLighter = NO;
+    cell.unselectedCellValueColor = [WPStyleGuide wordPressBlue];
+    
+    StatsSummary *todaySummary = self.sectionData[@(StatsSectionInsightsTodaysStats)];
+    
+    switch (row) {
+        case 1: // Views
+        {
+            cell.categoryIconLabel.text = @"";
+            cell.categoryLabel.text = [NSLocalizedString(@"Views", @"") uppercaseStringWithLocale:[NSLocale currentLocale]];
```
Comment (jleandroperez) [inline WordPressCom-Stats-iOS/InsightsTableViewController.m]: Maybe we could use a category method... `uppercaseLocaleString` perhaps! (Just thinking aloud, no need to do anything!)

File: WordPressCom-Stats-iOS/WPStatsViewController.m [hunk 4]
```diff
@@ -63,7 +63,7 @@ - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
     }
 }
 
-#pragma mark UIViewController overrides
+#pragma mark - UIViewController overrides
```
Comment (jleandroperez) [inline WordPressCom-Stats-iOS/WPStatsViewController.m]: YAY for `#pragma mark -`, i'm doing the exact same thing all over!

**Comment by jleandroperez** [comment]
@astralbodies code looks great!. I've tested this on iPad + iPhone, and verified that the stats are effectively matching with what's displayed on the web.

:shipit: on my side (please, hold for confirmation from Danilo + Jan!)

**Comment by astralbodies** [comment]
@daniloercoli Yeah for now until I can get size classes to work right on iOS when we drop iOS 7 it's the same layout for both orientations on iPhone.

**Comment by jancavan** [comment]
@astralbodies I don't see those dashes that appear when numbers are being loaded? But other than that, everything else looks good.

img_2337

Probably unrelated, but only Days is loading for me?
img_2338

**Comment by astralbodies** [comment]
Good eye @jancavan - we don't do the dashes (yet) on the other tabs which is where I borrowed the logic from. I'll make sure to update that and will push StatsDemo with the fix tomorrow.

**Comment by astralbodies** [comment]
Thanks everyone for the reviews!

**Pull request merged**